File: DSA/AYUSHMEHTA_590015131/Day13_Question2.cpp
#include <iostream>
#include <unordered_map>
#include <string>
using namespace std;

bool canFormPalindrome(const string& s) {
    unordered_map<char, int> freq;

    for (char c : s) {
        freq[c]++;
    }

    int oddCount = 0;
    for (auto pair : freq) {
        if (pair.second % 2 == 1) {
            oddCount++;
        }
    }

    return oddCount <= 1;
}

int main() {
    string s;
    cout << "Enter a string: ";
    cin >> s;

    if (canFormPalindrome(s)) {
        cout << "Yes, a permutation can form a palindrome.\n";
    } else {
        cout << "No, it cannot form a palindrome.\n";
    }

    return 0;
}

File: DSA/VaniJain_590012020/Day13Question2.java
import java.util.*;
public class Main 
{
    public static void main(String[] args) 
    {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a string: ");
        String s = scanner.nextLine();
        int[] freq = new int[128]; 
        for (int i = 0; i < s.length(); i++) 
        {
            char c = s.charAt(i);
            freq[c]++;
        }
        int oddCount = 0;
        for (int i = 0; i < 128; i++) 
        {
            if (freq[i] % 2 != 0) 
            {
                oddCount++;
            }
        }
        if (oddCount <= 1) 
        {
            System.out.println("Can form a palindrome");
        } else 
        {
            System.out.println("Cannot form a palindrome");
        }
    }
}
